substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1203,11 +1204,30 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
         }
 
         int exitStatus = 1;
+        File argsFile = null;
         try {
+            if (OS.getCurrent() == OS.WINDOWS) {
+                int totalLength = command.stream().mapToInt(s -> s.length() + 1).sum();
+                if (totalLength >= WINDOWS_MAX_COMMAND_LENGTH) {
+                    argsFile = File.createTempFile("native-image", "args");
+                    argsFile.deleteOnExit();
+                    Files.write(argsFile.toPath(), (Iterable<String>) command.stream().skip(1)::iterator);
```

Comment:
Did you verify that this works if the arguments contain spaces (e.g. in paths)?
See https://docs.oracle.com/en/java/javase/14/docs/specs/man/java.html#java-command-line-argument-files

Reply:
Looking at https://docs.oracle.com/en/java/javase/14/docs/specs/man/java.html#java-command-line-argument-files shows that some care is needed to ensure we do not run into problems with quoting and character escaping.

Reply:
OK, arguments that have spaces are now quoted (not sure we need to check for other whitespace chars as well? Given the fact it's almost all generated instead of user-supplied?) and backspaces are duplicated correctly.

Reply:
I'll this through all our gates and let you know if I find any quoting/escaping-related issues.

Reply:
All internal CI gates are passing with a57da3f553aaec8ca353778bb5f24ccabff2fea1.